l2gersync/evm_downloader_sovereign_test.go

```diff
@@ -0,0 +1,113 @@
+package l2gersync
+
+import (
+	"context"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/agglayer/aggkit/l1infotreesync"
+	l2gersyncmocks "github.com/agglayer/aggkit/l2gersync/mocks"
+	"github.com/agglayer/aggkit/sync"
+	aggkittypes "github.com/agglayer/aggkit/types"
+	aggkittypesmocks "github.com/agglayer/aggkit/types/mocks"
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDownloaderSovereign_Download(t *testing.T) {
+	t.Parallel()
+
+	fromBlock := uint64(100)
+	syncBlockChunkSize := uint64(10)
+	latestBlock := uint64(120)
+	l2GERAddr := common.HexToAddress("0x1234567890abcdef1234567890abcdef12345678")
+
+	mockL2Client := aggkittypesmocks.NewBaseEthereumClienter(t)
+	mockL1InfoTreeSync := &l2gersyncmocks.L1InfoTreeQuerier{}
+	rh := &sync.RetryHandler{
+		MaxRetryAttemptsAfterError: 5,
+		RetryAfterErrorPeriod:      time.Millisecond,
+	}
+
+	testGER := common.HexToHash("0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef")
+	testHashChainValue := common.HexToHash("0xabcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890")
+	testL1InfoTreeIndex := uint32(42)
+	testBlockHeader := &ethtypes.Header{
+		Number:      big.NewInt(int64(fromBlock)),
+		ParentHash:  common.HexToHash("0xabc123"),
+		Root:        common.HexToHash("0xdef456"),
+		TxHash:      common.HexToHash("0x789abc"),
+		ReceiptHash: common.HexToHash("0x101112"),
+		Time:        uint64(time.Now().Unix()),
+		GasLimit:    8000000,
+		GasUsed:     21000,
+	}
+	testBlockHash := testBlockHeader.Hash()
+	testLogs := []ethtypes.Log{
+		{
+			Address:     l2GERAddr,
+			Topics:      []common.Hash{insertGEREventSignature, testGER, testHashChainValue},
+			Data:        []byte{},
+			BlockNumber: fromBlock,
+			TxHash:      common.HexToHash("0x111"),
+			TxIndex:     0,
+			BlockHash:   testBlockHash,
+			Index:       0,
+		},
+	}
+
+	mockL2Client.On("ChainID", mock.Anything).Return(big.NewInt(1), nil).Maybe()
```

Comment:
The approach with using expected is better, because it would produce a compile time error, in case function gets renamed, removed etc.

I would suggest changing it on the other places within this test as well.

```suggestion
	mockL2Client.EXPECT().ChainID(mock.Anything).Return(big.NewInt(1), nil).Maybe()
```